cmd/bee-split/main.go

```diff
@@ -0,0 +1,241 @@
+	baseUrl string
+}
+
+// newApiStore creates a new apiStor
+func newApiStore(host string, port int, ssl bool) (storage.Putter, error) {
+	scheme := "http"
+	if ssl {
+		scheme += "s"
+	}
+	u := &url.URL{
+		Host:   fmt.Sprintf("%s:%d", host, port),
+		Scheme: scheme,
+		Path:   "bzz-chunk",
+	}
+	return &apiStore{
+		baseUrl: u.String(),
+	}, nil
+}
+
+// Put implements storage.Putter
+func (a *apiStore) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chunk) (exist []bool, err error) {
+	for _, ch := range chs {
+		addr := ch.Address().String()
+		buf := bytes.NewReader(ch.Data())
+		url := strings.Join([]string{a.baseUrl, addr}, "/")
+		c := &http.Client{}
+		res, err := c.Post(url, "application/octet-stream", buf)
+		if err != nil {
+			return []bool{}, err
+		}
+		_ = res
+	}
+	return []bool{}, nil
+}
+
+// limitReadCloser wraps the input to the application to limit the input to the given count flag.
+type limitReadCloser struct {
+	io.Reader
+	closeFunc func() error
+}
+
+// newLimitReadCloser creates a new limitReadCloser.
+func newLimitReadCloser(r io.Reader, closeFunc func() error, c int64) io.ReadCloser {
+	return &limitReadCloser{
+		Reader:    io.LimitReader(r, c),
+		closeFunc: closeFunc,
+	}
+}
+
+// Close implements io.Closer
+func (l *limitReadCloser) Close() error {
+	return l.closeFunc()
+}
+
+// Split is the underlying procedure for the CLI command
+func Split(cmd *cobra.Command, args []string) (err error) {
+	var infile io.ReadCloser
+
+	// if one arg is set, this is the input file
+	// if not, we are reading from standard input
+	if len(args) > 0 {
+
+		// get the file length
+		info, err := os.Stat(args[0])
+		if err != nil {
+			return err
+		}
+		fileLength := info.Size()
+
+		// check if we are limiting the input, and if the limit is valid
+		if inputLength > 0 {
+			if inputLength > fileLength {
+				return fmt.Errorf("input data length set to %d on file with length %d", inputLength, fileLength)
+			}
+		} else {
+			inputLength = fileLength
+		}
+
+		// open file and wrap in limiter
+		f, err := os.Open(args[0])
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		infile = newLimitReadCloser(f, f.Close, inputLength)
+	} else {
+		// this simple splitter is too stupid to handle open-ended input, sadly
+		if inputLength == 0 {
+			return errors.New("must specify length of input on stdin")
+		}
+		infile = newLimitReadCloser(os.Stdin, func() error { return nil }, inputLength)
+	}
+
+	// add the fsStore and/or apiStore, depending on flags
+	stores := newTeeStore()
+	if outdir != "" {
+		err := os.MkdirAll(outdir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+		store := newFsStore(outdir)
+		stores.Add(store)
+	}
+	if !noHttp {
+		store, err := newApiStore(host, port, ssl)
+		if err != nil {
+			return err
+		}
+		stores.Add(store)
+	}
+
+	// split and rule
+	s := splitter.NewSimpleSplitter(stores)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
```

Comment:
since you're not using any cobra context and i can't see any code that listens to system signals, so the question asked is - can you even `kill` this process at all? if not - is this the intended behavior?

Reply:
`SIGINT` will terminate it